doc/Conan.md

```diff
@@ -0,0 +1,97 @@
+### For new members
+It is common practice to surround optional arguments to a command in square braces []. Thus we will try to use the same formatting in our brief documentation. Despite this please use the options prescribed by us in the docs unless you understand what the command will do without that option.
+
+
+## Who is this Document for?
+This document is primarily geared towards both new and old members who are working between both mtrain-firmware and robocup-firmware and thus need to know how to manage the package created by building mtrain firmware as well as those who would like a deeper understanding of what conan is doing for us in terms of package management.
+It is not intended as a full guide to Conan as it only walks through the basic steps of updating a package for and changing requires but I will leave links to relevant documentation and give commentary as I go.
+
+
+## Conan
+Conan is a decentralized c++ package manager. This means that much like git it can work it multiple remotes to obtain the dependencies it requires and it can upload built packages to these remotes. The remotes are merely just package storage and do not build the packages themselves.
+The default location for your conan directory will be ~/.conan/ . Here conan stores all user configurations and packages (under the data directory). It is ill advised to modify any of these files by hand but it is sometimes helpful to force rebuild packages by deleting the contents of the data directory.
+
+To learn more about Conan see the introduction page on the official docs:
+https://docs.conan.io/en/latest/introduction.html
+
+## Dependency Management
+TODO talk more about how conan deals with dependencies eg requires statement
+
+## Binary Management
+Conan also is capable of managing binaries targeted at specific platforms within these binaries. The settings for building for a specific platform are described in a conan profile which are stored in ~/.conan/profiles/ . For robocup-firmware you will be using the will be using the armv7hf profile to build for the mtrain platform.
+
+
+## Creating a local conan package:
+For those who are working on both mtrain and robocup firmware you may want to have seperate testing version of the mtrain package to do so would make a package with a different reference (which makes the package into a different folder in your ~/.conan/data/ folder) and then change the package dependencies in robocup-firmware to match the new package reference.
+
+The below is used to create a local version of the conan package:
+```
+conan create . collin/testing -pr armv7hf
```

Comment:
There's usually some normal way to say "put your name here" in commands, not quite sure

Reply:
Usually you would do <yourname> . That or [yourname] or something of the sort. I kinda mentioned at the top that [ ] means optional but didnt really continue to put it in every case it was true. I am always afraid of doing this as many 2200 students will put their names in angle brackets and then ask they the command we gave them failed. I may just change it to yourname/testing

Reply:
Ooo, yeah, maybe not then. This was definitely on the bottom of things that might cause problems. Do whatever works in 2200